extensions/ql-vscode/src/view/variant-analysis/VariantAnalysis.tsx

```diff
@@ -60,11 +61,16 @@ export function VariantAnalysis({
   const [repoResults, setRepoResults] =
     useState<VariantAnalysisScannedRepositoryResult[]>(initialRepoResults);
 
-  const [selectedRepositoryIds, setSelectedRepositoryIds] = useState<number[]>(
-    [],
-  );
+  const [selectedRepositoryIds, setSelectedRepositoryIds] =
+    useStateWithTelemetry<number[]>(
+      [],
+      "variant-analysis-selected-repository-ids",
+    );
   const [filterSortState, setFilterSortState] =
-    useState<RepositoriesFilterSortState>(defaultFilterSortState);
+    useStateWithTelemetry<RepositoriesFilterSortState>(
+      defaultFilterSortState,
+      "variant-analysis-filter-sort-state",
```

Comment:
This will potentially result in a lot of these events being send. I believe that if you start searching for "github", you will get 6 events, 1 for every keystroke. I'm not sure if that's the intention here.

Reply:
That's not really the intention. Perhaps it would be worth adding a debounce here or something else to limit the number of telemetry events emitted. I don't think we want this to affect the call to `useState`, since it's currently working ok and I don't want to introduce more complexity unnecessarily, but we could have the telemetry be rate limited.

Reply:
I've now implemented this as an option to `useTelemetryOnChange`.